Description: Fix CONFIG_INFINIBAND_IPOIB_CM 
 From linux efc1eed:
 .
 If the above is turned off then ipoib_cm_dev_init unconditionally
 returns ENOSYS, and the newly added error handling in
 0b3957 prevents ipoib from coming up at all:
 .
 kernel: mlx4_0: ipoib_transport_dev_init failed
 kernel: mlx4_0: failed to initialize port 1 (ret = -12).
Author: Brian T. Smith <bsmith@systemfabricworks.com>
Author: Jason Gunthorpe <jgunthorpe@obsidianresearch.com>
Origin: https://github.com/torvalds/linux/commit/efc1eedbf63a194b3b576fc25776f3f1fa55a4d4#diff-354653dff7ec9ddd6ac73f85f2956316
Forwarded: not-needed
Last-Update: <2017-12-14>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/ib_ipoib/ipoib_verbs.c
+++ b/ib_ipoib/ipoib_verbs.c
@@ -176,7 +176,8 @@
 		else
 			size += ipoib_recvq_size * ipoib_max_conn_qp;
 	} else
-		goto out_free_wq;
+		if (ret != -ENOSYS)
+			goto out_free_wq;
 
 	priv->recv_cq = ib_create_cq(priv->ca, ipoib_ib_completion, NULL, dev, size, 0);
 	if (IS_ERR(priv->recv_cq)) {
